src/uct/base/uct_iface.c

```diff
@@ -242,6 +242,27 @@ static int uct_iface_is_same_device(const uct_iface_h iface,
     return !memcmp(device_addr, dev_addr, attr.device_addr_len);
 }
 
+void uct_iface_populate_info_str_buf(
+                                const uct_iface_is_reachable_params_t *params,
+                                const char *fmt, ...)
+{
+    char *info_str_buf = UCS_PARAM_VALUE(UCT_IFACE_IS_REACHABLE_FIELD,
+                                    params, info_string, INFO_STRING,
+                                    NULL);
+    size_t info_str_buf_len = UCS_PARAM_VALUE(UCT_IFACE_IS_REACHABLE_FIELD,
+                                        params, info_string_length,
+                                        INFO_STRING_LENGTH, 0);
+
+    va_list args;
+    va_start(args, fmt);
+
+    if (info_str_buf && info_str_buf_len > 0 && info_str_buf[0] == '\0') {
```

Comment:
1. pls use ( ) around internal expressions
2. why need to check info_str_buf[0] == '\0' ?
3. use (info_str_buf != NULL) - check for NULL explicitly
4. probably no need to check info_str_buf_len > 0, it's already handled by ucs_snprintf_safe